instrumentation/kafka-streams/src/test/java/brave/kafka/streams/BaseTracingTest.java

```diff
@@ -0,0 +1,176 @@
+                        }
+
+                        @Override
+                        public <K, V> void forward(K key, V value, String childName) {
+                        }
+
+                        @Override
+                        public void commit() {
+                        }
+
+                        @Override
+                        public String topic() {
+                            return TEST_TOPIC;
+                        }
+
+                        @Override
+                        public int partition() {
+                            return 0;
+                        }
+
+                        @Override
+                        public long offset() {
+                            return 0;
+                        }
+
+                        @Override
+                        public Headers headers() {
+                            return headers;
+                        }
+
+                        @Override
+                        public long timestamp() {
+                            return 0;
+                        }
+
+                        @Override
+                        public Map<String, Object> appConfigs() {
+                            return null;
+                        }
+
+                        @Override
+                        public Map<String, Object> appConfigsWithPrefix(String prefix) {
+                            return null;
+                        }
+                    };
+
+    ConcurrentLinkedDeque<Span> spans = new ConcurrentLinkedDeque<>();
+    Tracing tracing = Tracing.newBuilder()
+            .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+                    .addScopeDecorator(StrictScopeDecorator.create())
+                    .build())
+            .spanReporter(spans::add)
+            .build();
+    KafkaStreamsTracing kafkaStreamsTracing = KafkaStreamsTracing.create(tracing);
+
+    ProcessorSupplier<String, String> fakeProcessorSupplier =
+            kafkaStreamsTracing.processorSupplier(
+                    "processor-1",
+                    new AbstractProcessor<String, String>() {
+                        @Override
+                        public void process(String key, String value) {
+                            context().forward(key, value);
+                        }
+                    });
+
+    TransformerSupplier<String, String, String> fakeTransformerSupplier =
+            kafkaStreamsTracing.transformerSupplier(
+                    "transformer-1",
+                    new Transformer<String, String, String>() {
+                        ProcessorContext context;
+                        @Override
+                        public void init(ProcessorContext context) {
+                            this.context = context;
+                        }
+
+                        @Override
+                        public String transform(String key, String value) {
+                            return "transformed";
```

Comment:
maybe return value? then it is more like a typical transform?

Reply:
Agree